core/src/test/java/org/elasticsearch/index/query/GeoBoundingBoxQueryBuilderTest.java

```diff
@@ -0,0 +1,346 @@
+        GeoBoundingBoxQueryBuilder builder = new GeoBoundingBoxQueryBuilder("mapped_geo");
+        Rectangle box = RandomShapeGenerator.xRandomRectangle(getRandom(), RandomShapeGenerator.xRandomPoint(getRandom()));
+
+        if (frequently()) {
+            // check the top-left/bottom-right combination of setters
+            int path = randomIntBetween(0, 2);
+            switch (path) {
+            case 0:
+                builder.topLeft(new GeoPoint(box.getMaxY(), box.getMinX()));
+                break;
+            case 1:
+                builder.topLeft(GeohashUtils.encodeLatLon(box.getMaxY(), box.getMinX()));
+                break;
+            default:
+                builder.topLeft(box.getMaxY(), box.getMinX());
+            }
+
+            path = randomIntBetween(0, 2);
+            switch (path) {
+            case 0:
+                builder.bottomRight(new GeoPoint(box.getMinY(), box.getMaxX()));
+                break;
+            case 1:
+                builder.bottomRight(GeohashUtils.encodeLatLon(box.getMinY(), box.getMaxX()));
+                break;
+            default:
+                builder.bottomRight(box.getMinY(), box.getMaxX());
+            }
+        } else {
+            // check the deprecated top-right/bottom-left combination of setters
+            int path = randomIntBetween(0, 2);
+            switch (path) {
+            case 0:
+                builder.topRight(new GeoPoint(box.getMaxY(), box.getMaxX()));
+                break;
+            case 1:
+                builder.topRight(GeohashUtils.encodeLatLon(box.getMaxY(), box.getMaxX()));
+                break;
+            default:
+                builder.topRight(box.getMaxY(), box.getMaxX());
+            }
+
+            path = randomIntBetween(0, 2);
+            switch (path) {
+            case 0:
+                builder.bottomLeft(new GeoPoint(box.getMinY(), box.getMinX()));
+                break;
+            case 1:
+                builder.bottomLeft(GeohashUtils.encodeLatLon(box.getMinY(), box.getMinX()));
+                break;
+            default:
+                builder.bottomLeft(box.getMinY(), box.getMinX());
+            }
+        }
+
+        if (randomBoolean()) {
+            builder.coerce(randomBoolean());
+        }
+        if (randomBoolean()) {
+            builder.ignoreMalformed(randomBoolean());
+        }
+
+        builder.type(randomFrom(Type.values()));
+        return builder;
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testValidation_empty_fieldname() {
+        new GeoBoundingBoxQueryBuilder("");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testValidation_null_fieldname() {
+        new GeoBoundingBoxQueryBuilder(null);
+    }
+
+    @Test
+    @Override
+    public void testToQuery() throws IOException {
+        QueryShardContext context = createShardContext();
```

Comment:
I think this is all because there are times when we run the tests without registered types. Then we would just need an assume here: `assumeTrue("test runs only when at least a type is registered", getCurrentTypes().length > 0);`. no need for doToQuery etc. just call super.testToQuery(). And test the exception thrown when there's no types separately (if we don't do that already)?